pkg/armhelpers/storage.go

```diff
@@ -43,9 +44,29 @@ func (az *AzureClient) getStorageKeys(ctx context.Context, resourceGroup, accoun
 // TODO(colemick): why doesn't SDK give a way to just delete a blob by URI?
 // it's what it ends up doing internally anyway...
 func (as *AzureStorageClient) DeleteBlob(vhdContainer, vhdBlob string) error {
-	bs := as.client.GetBlobService()
-	containerRef := bs.GetContainerReference(vhdContainer)
+	containerRef := getContainerRef(as, vhdContainer)
 	blobRef := containerRef.GetBlobReference(vhdBlob)
 
 	return blobRef.Delete(&azStorage.DeleteBlobOptions{})
 }
+
+// CreateContainer creates the CloudBlobContainer if it does not exist
+func (as *AzureStorageClient) CreateContainer(containerName string) (bool, error) {
+	containerRef := getContainerRef(as, containerName)
+	created, err := containerRef.CreateIfNotExists(nil)
```

Comment:
This needs to have ContainerOptions as a parameter. Let's add flexibility and agility here